src/main/java/io/camunda/zeebe/bpmnassert/testengine/RecordStreamSource.java

```diff
@@ -0,0 +1,93 @@
+package io.camunda.zeebe.bpmnassert.testengine;
+
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.RecordValue;
+import io.camunda.zeebe.protocol.record.ValueType;
+import io.camunda.zeebe.protocol.record.value.*;
+import io.camunda.zeebe.protocol.record.value.deployment.Process;
+import io.camunda.zeebe.test.util.record.CompactRecordLogger;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+public interface RecordStreamSource {
+
+  Iterable<Record<?>> records();
+
+  default <T extends RecordValue> Iterable<Record<T>> recordsOfValueType(
+      final ValueType valueType) {
+    final Stream<Record<T>> stream =
+        (Stream)
+            StreamSupport.stream(records().spliterator(), false)
+                .filter((record) -> record.getValueType() == valueType);
+    return stream::iterator;
+  }
+
+  default Iterable<Record<ProcessInstanceRecordValue>> processInstanceRecords() {
+    return recordsOfValueType(ValueType.PROCESS_INSTANCE);
+  }
+
+  default Iterable<Record<JobRecordValue>> jobRecords() {
+    return recordsOfValueType(ValueType.JOB);
+  }
+
+  default Iterable<Record<JobBatchRecordValue>> jobBatchRecords() {
+    return recordsOfValueType(ValueType.JOB_BATCH);
+  }
+
+  default Iterable<Record<DeploymentRecordValue>> deploymentRecords() {
+    return recordsOfValueType(ValueType.DEPLOYMENT);
+  }
+
+  default Iterable<Record<Process>> processRecords() {
+    return recordsOfValueType(ValueType.PROCESS);
+  }
+
+  default Iterable<Record<VariableRecordValue>> variableRecords() {
+    return recordsOfValueType(ValueType.VARIABLE);
+  }
+
+  default Iterable<Record<VariableDocumentRecordValue>> variableDocumentRecords() {
+    return recordsOfValueType(ValueType.VARIABLE_DOCUMENT);
+  }
+
+  default Iterable<Record<IncidentRecordValue>> incidentRecords() {
+    return recordsOfValueType(ValueType.INCIDENT);
+  }
+
+  default Iterable<Record<TimerRecordValue>> timerRecords() {
+    return recordsOfValueType(ValueType.TIMER);
+  }
+
+  default Iterable<Record<MessageRecordValue>> messageRecords() {
+    return recordsOfValueType(ValueType.MESSAGE);
+  }
+
+  default Iterable<Record<MessageSubscriptionRecordValue>> messageSubscriptionRecords() {
+    return recordsOfValueType(ValueType.MESSAGE_SUBSCRIPTION);
+  }
+
+  default Iterable<Record<MessageStartEventSubscriptionRecordValue>>
+      messageStartEventSubscriptionRecords() {
+    return recordsOfValueType(ValueType.MESSAGE_START_EVENT_SUBSCRIPTION);
+  }
+
+  default Iterable<Record<ProcessMessageSubscriptionRecordValue>>
+      processMessageSubscriptionRecords() {
+    return recordsOfValueType(ValueType.PROCESS_MESSAGE_SUBSCRIPTION);
+  }
+
+  default void print(final boolean compact) {
```

Comment:
:wrench: Maybe add `toString()`method that calls print

Reply:
`RecordStreamSource` is an interface, so I can not add this here. I'd have to add it to `ZeebeEngineImpl` but it doesn't feel right to have the `toString` of the complete engine only print out the records.

Reply:
You can add a `default toString()` 

And I think I had a different comment that in my mind `ZeebeEngine` should not implement `RecordStreamSource` but rather have a getter to return a record stream source. So those two go hand in hand, in my mind.

Reply:
It's still a bit awkward to do this. The purpose of `print()` is to write the log to the console. It doesn't return anything. `toString()` should be a string representation of the object. I think returning a string containing all log records is a bit excessive.